TypeCobol/Compiler/CupParser/NodeBuilder/ProgramClassBuilder.cs

```diff
@@ -1341,6 +1341,7 @@ public virtual void StartWhenConditionClause(List<TypeCobol.Compiler.CodeElement
 
         public virtual void EndWhenConditionClause()
         {
+            WhenConditionNodeChecker.OnNode(SyntaxTree.CurrentNode);
```

Comment:
It seems that no, I tried and I got that other tests no longer pass, but also here is a syntactical error it should be reported during the parsing phase.

Reply:
What do you mean by "parsing phase"? ExecutionStep.SemanticCheck?
Our architecture rules state that every check of this kind are done in a checker during CrossCheck step (ExecutionStep.CrossCheck ). Why would we need to make an exception in this case?

Reply:
This should go there, right ?

https://github.com/TypeCobolTeam/TypeCobol/blob/4cc51f941a01dbb62eb94ba0f2db5330d22ffc00/TypeCobol/Compiler/Diagnostics/CrossChecker.cs#L471

Reply:
Or directly on `Visit(Evaluate)` because we don't need to do this check on every `when`.
We only need to check that the last `when` different than a `when other` and the `when other` contains at least one instruction.
And I just notice that there is no test for all cases.
We could also check `search`/`search all` but it's a little out of the scope of this issue. I can write an issue about that.

Reply:
I've created issue #2087 and #2088 for similar problems, but we can only focus on the evaluate right now.

Reply:
What I am saying is that when  I move this check in CrossChecker I got unexpected results.